stresser2/workers.go

```diff
@@ -87,70 +97,104 @@ func genKeyTracker() *keyTracker {
+			for j := 0; j < config.RBatchSize; j++ {
+				kList[j] = KeyTracker.getRKey()
+
+				if config.HashedKeys {
+					kList[j] = h256(kList[j])
+				}
 			}
 			_, err = client.GetAll(ctx, kList)
 		}
 
 		if err != nil && !strings.Contains(err.Error(), "key not found") {
 			log.Fatalln("Failed to read the batch. Reason:", err)
 		} else {
-			counter += int64(config.BatchSize)
-			atomic.AddInt64(totalCounter, int64(config.BatchSize))
+			counter += int64(config.RBatchSize)
+			atomic.AddInt64(totalCounter, int64(config.RBatchSize))
 		}
 	}
-	elapsed=float64(time.Since(t0).Seconds())
-	log.Printf("%s DONE: read %d entries in %.3fs, %.3f KV/s", jobid, counter, elapsed, float64(counter)/float64(time.Since(t0).Seconds()))
+
+	elapsed = time.Since(t0)
+
+	log.Printf("%s DONE: read %d entries in %v, %.3f KV/s", jobid, counter, elapsed, float64(counter)/elapsed.Seconds())
+
 	return counter, elapsed
 }
 
-func writeWorker(n int, totalCounter *int64) (counter int64, elapsed float64) {
+func writeWorker(n int, totalCounter *int64) (counter int64, elapsed time.Duration) {
 	jobid := fmt.Sprintf("WJOB%02d", n)
 
 	ctx, client := connect(jobid)
 	defer client.CloseSession(ctx)
+
+	var t1 time.Time
+
+	kvList := &schema.SetRequest{KVs: make([]*schema.KeyValue, config.WBatchSize), NoWait: config.RWorkers == 0}
+
 	t0 := time.Now()
-	t1 := time.Now()
+
 	for i := 0; i < config.WBatchNum; i++ {
 		t1 = time.Now()
 
-		kvs := make([]*schema.KeyValue, config.BatchSize)
+		for j := 0; j < config.WBatchSize; j++ {
+			k := KeyTracker.getWKey()
+
+			if config.HashedKeys {
+				k = h256(k)
+			}
+
+			v := make([]byte, config.PayloadSize)
 
-		for j := 0; j < config.BatchSize; j++ {
-			kvs[j] = &schema.KeyValue{
-				Key:   h256(KeyTracker.getWKey()),
-				Value: getPayload(256),
+			if config.RandomPayloads {
+				rnd.Read(v)
+			}
+
+			kvList.KVs[j] = &schema.KeyValue{
+				Key:   k,
+				Value: v,
 			}
 		}
 
-		kvList := &schema.SetRequest{KVs: kvs, NoWait: true}
 		if _, err := client.SetAll(ctx, kvList); err != nil {
 			log.Fatalln("Failed to submit the batch. Reason:", err)
 		} else {
-			counter += int64(config.BatchSize)
-			atomic.AddInt64(totalCounter, int64(config.BatchSize))
+			counter += int64(config.WBatchSize)
+			atomic.AddInt64(totalCounter, int64(config.WBatchSize))
 		}
+
 		if config.WSpeed > 0 {
 			dt := time.Since(t1)
-			sleepTime := (1000.0 * float64(config.BatchSize) / float64(config.WSpeed)) - float64(dt.Milliseconds())
+			sleepTime := (1000.0 * float64(config.WBatchSize) / float64(config.WSpeed)) - float64(dt.Milliseconds())
 			time.Sleep(time.Duration(sleepTime) * time.Millisecond)
 		}
 
 	}
-	elapsed=float64(time.Since(t0).Seconds())
-	log.Printf("%s DONE: inserted %d entries in %.3fs, %.3f KV/s", jobid, counter, elapsed, float64(counter)/float64(time.Since(t0).Seconds()))
+
+	elapsed = time.Since(t0)
+
+	log.Printf("%s DONE: inserted %d entries in %v, %.3f KV/s", jobid, counter, elapsed, float64(counter)/elapsed.Seconds())
```

Comment:
would be good to also print TX/s

Reply:
Note that this is just the partial output for one worker.
Total number of TX/s is already printed on line 206 of main.go.
Nevertheless, I've added TX/s figure in commit d263085